doc/topics/cloud/vmware.rst

```diff
@@ -0,0 +1,163 @@
+===================================
+Getting Started With VMware vSphere
+===================================
+
+VMware vSphere is a server virtualization platform for building cloud
+infrastructure.
+
+
+Dependencies
+============
+The vmware module for Salt Cloud requires the ``pyVmomi`` package, which is
+available at PyPI:
+
+https://pypi.python.org/pypi/pyvmomi
+
+This package can be installed using `pip` or `easy_install`:
+
+.. code-block:: bash
+
+    pip install pyvmomi
+    easy_install pyvmomi
+
+
+Configuration
+=============
+The VMware cloud module needs the vCenter URL, username and password to be
+set up in the cloud configuration at
+``/etc/salt/cloud.providers`` or ``/etc/salt/cloud.providers.d/vmware.conf``:
+
+.. code-block:: yaml
+
+    my-vmware-config:
+      provider: vmware
+      user: DOMAIN\user
+      password: verybadpass
+      url: vcenter01.domain.com
+
+    vmware-vcenter02:
+      provider: vmware
+      user: DOMAIN\user
+      password: verybadpass
+      url: vcenter02.domain.com
+
+    vmware-vcenter03:
+      provider: vmware
+      user: DOMAIN\user
+      password: verybadpass
+      url: vcenter03.domain.com
+
+
+Profiles
+========
+Set up an initial profile at ``/etc/salt/cloud.profiles`` or
+``/etc/salt/cloud.profiles.d/vmware.conf``:
+
+.. code-block:: yaml
+
+    vmware-centos6.5:
+      provider: vmware-vcenter01
+      clonefrom: test-vm
```

Comment:
The plan is to deprecate the vsphere driver in the future release and use vmware driver instead since it uses the official python library that VMware suggests. Also, I've tested both and vmware driver is faster and more efficient that the vsphere driver.